src/CsvHelper/CsvParser.cs

```diff
@@ -858,10 +950,12 @@ protected ProcessedField ProcessRFC4180Field(int start, int length, int quoteCou
 				// Not quoted.
 				// No processing needed.
 
-				return new ProcessedField(newStart, newLength, buffer);
+				return new ProcessedField(newStart, GetActualFieldLength(newLength), buffer);
 			}
 
-			if (buffer[newStart] != quote || buffer[newStart + newLength - 1] != quote || newLength == 1 && buffer[newStart] == quote)
+			if (buffer[newStart] != quote ||

```

Comment:
Maybe we could move these conditions to an additional method and call it, I think we can make the purpose of these conditions more clear